### DISCUSSION TITLE
Docker Permissions

### DISCUSSION THREAD

**Discussion by csbuja** [pytest-dev/pytest#9068] [Q&A]
https://stackoverflow.com/questions/64138038/docker-sqlite3-operationalerror-unable-to-open-database-file

Cross posting from Stack Overflow. What is the name of the database used in this post, so I can change the permissions for this DB file? My plan is to run chmod on the db file in my Dockerfile.

Thanks in advance!

**Comment by portyu9** [comment]
Hi @csbuja 

The database in that traceback is **coverage.py's data file**, not an application database.

Coverage stores its data in a SQLite file whose default name is:

```text
.coverage
```

So:

```text
sqlite3.OperationalError: unable to open database file
```

means the user running pytest/pytest-cov cannot create or write the coverage data file at the selected location.

The better Docker fix is to make the working/data directory writable by the runtime user, rather than trying to `chmod` a mystery database file after the image is built.

You can also explicitly put coverage data somewhere writable:

```dockerfile
ENV COVERAGE_FILE=/tmp/.coverage
```

or configure coverage's `[run] data_file`.

If the project directory is intended to be writable, make its ownership match your container user instead:

```dockerfile
RUN chown -R appuser:appuser /workspace
USER appuser
```

pytest itself does not own this database behavior; pytest-cov delegates coverage collection/storage to coverage.py.

Please mark this answer as accepted if it helped, thank you.
